File: army_challenge/services/army_statistics.py
from .types import IArmy, IUnitType, IUnit

class ArmyStatistics:

    @staticmethod
    def set_army_strength(army:IArmy)
        army.armyStrength= self.getUnitsStrength(army.units.pikemans) +
            self.getUnitsStrength(army.units.archers) +
            self.getUnitsStrength(army.units.knights)

    @staticmethod
    def getUnitsStrength<T extends IUnitType> (unit:IUnit<T>[]): int
        return unit.reduce((acc:int, unit:IUnit<T>)=>
            return acc += unit.strength as int
        ,0)

File: src/services/armyStatistics.ts
import type {
  IArmy,
  IUnitType,
  IUnit
} from '../types/index';
export class ArmyStatistics {

    static setArmyStrength(army:IArmy) {
        army.armyStrength= this.getUnitsStrength(army.units.pikemans) +
            this.getUnitsStrength(army.units.archers) +
            this.getUnitsStrength(army.units.knights);
    }

    static getUnitsStrength<T extends IUnitType> (unit:IUnit<T>[]): number {
        return unit.reduce((acc:number, unit:IUnit<T>)=>{
            return acc += unit.strength as number
        },0)
    }

}